daemon/daemon_unix.go

```diff
@@ -268,8 +268,10 @@ func verifyContainerResources(resources *containertypes.Resources, sysInfo *sysi
 	warnings := []string{}
 
 	// memory subsystem checks and adjustments
-	if resources.Memory != 0 && resources.Memory < linuxMinMemory {
-		return warnings, fmt.Errorf("Minimum memory limit allowed is 4MB")
+	if resources.Memory != 0 {
+		if resources.Memory != -1 && resources.Memory < linuxMinMemory {
```

Comment:
I just wanted to be logically separated. since it's all `and` could be in same line though. I don't have a strong opinion on either one.